File: src/main/java/no/nav/fo/veilarbregistrering/domain/OrdinaerBrukerRegistrering.java
package no.nav.fo.veilarbregistrering.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.experimental.Accessors;
import no.nav.fo.veilarbregistrering.domain.besvarelse.Besvarelse;
import no.nav.fo.veilarbregistrering.domain.besvarelse.Stilling;
import no.nav.fo.veilarbregistrering.profilering.Profilering;

import java.time.LocalDateTime;
import java.util.List;

import static no.nav.fo.veilarbregistrering.domain.BrukerRegistreringType.ORDINAER;

@Data
@Accessors(chain = true)
@ToString
@EqualsAndHashCode
public class OrdinaerBrukerRegistrering extends BrukerRegistrering {
    long id;
    LocalDateTime opprettetDato;
    Besvarelse besvarelse;
    List<TekstForSporsmal> teksterForBesvarelse;
    Stilling sisteStilling;
    Profilering profilering;

    @Override
    public BrukerRegistreringType hentType() {
        return ORDINAER;
    }

}

File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/OrdinaerBrukerRegistrering.kt
package no.nav.fo.veilarbregistrering.registrering.bruker

import no.nav.fo.veilarbregistrering.besvarelse.Besvarelse
import no.nav.fo.veilarbregistrering.besvarelse.Stilling
import no.nav.fo.veilarbregistrering.profilering.Profilering
import no.nav.fo.veilarbregistrering.registrering.BrukerRegistreringType
import no.nav.fo.veilarbregistrering.registrering.manuell.Veileder
import java.time.LocalDateTime

data class OrdinaerBrukerRegistrering(
    override val id: Long = 0,
    val opprettetDato: LocalDateTime = LocalDateTime.now(),
    val besvarelse: Besvarelse,
    val teksterForBesvarelse: List<TekstForSporsmal>,
    val sisteStilling: Stilling,
    val profilering: Profilering? = null,
    override var manueltRegistrertAv: Veileder? = null,
) : BrukerRegistrering() {

    override fun hentType(): BrukerRegistreringType {
        return BrukerRegistreringType.ORDINAER
    }

    override fun toString(): String {
        return "OrdinaerBrukerRegistrering(id=$id, opprettetDato=$opprettetDato, besvarelse=$besvarelse, teksterForBesvarelse=$teksterForBesvarelse, sisteStilling=$sisteStilling, profilering=$profilering)"
    }

    companion object {
        @JvmStatic
        fun medProfilering(registrering: OrdinaerBrukerRegistrering, profilering: Profilering): OrdinaerBrukerRegistrering =
            OrdinaerBrukerRegistrering(
                registrering.id,
                registrering.opprettetDato,
                registrering.besvarelse,
                registrering.teksterForBesvarelse,
                registrering.sisteStilling,
                profilering
            )
    }
}